spec.html

```diff
@@ -11828,14 +11828,25 @@ <h1>Static Semantics: Early Errors</h1>
         <emu-grammar>PrimaryExpression : RegularExpressionLiteral</emu-grammar>
         <ul>
           <li>
-            It is a Syntax Error if BodyText of |RegularExpressionLiteral| cannot be recognized using the goal symbol |Pattern| of the ECMAScript RegExp grammar specified in <emu-xref href="#sec-patterns"></emu-xref>.
-          </li>
-          <li>
-            It is a Syntax Error if FlagText of |RegularExpressionLiteral| contains any code points other than `"g"`, `"i"`, `"m"`, `"s"`, `"u"`, or `"y"`, or if it contains the same code point more than once.
+            It is a Syntax Error if IsValidRegularExpressionLiteral(|RegularExpressionLiteral|) is *false*.
           </li>
         </ul>
       </emu-clause>
 
+      <emu-clause id="sec-primary-expression-regular-expression-literals-static-semantics-isvalidregularexpressionliteral" aoid="IsValidRegularExpressionLiteral">
+        <h1>Static Semantics: IsValidRegularExpressionLiteral ( _literal_ )</h1>
+        <p>The abstract operation IsValidRegularExpressionLiteral determines if its argument is a valid regular expression literal. The following steps are taken:</p>
+        <emu-alg>
+          1. Assert: _literal_ is a |RegularExpressionLiteral|.
+          1. If FlagText of _literal_ contains any code points other than `"g"`, `"i"`, `"m"`, `"s"`, `"u"`, or `"y"`, or if it contains the same code point more than once, return *false*.
+          1. Let _P_ be BodyText of _literal_.
+          1. If FlagText of _literal_ contains `"u"`,
+            1. If _P_ cannot be recognized using the goal symbol |Pattern[+U, +N]| of the ECMAScript RegExp grammar specified in <emu-xref href="#sec-patterns"></emu-xref>, return *false*.
```

Comment:
This should probably include the stuff about elements being left over or Early Errors.
```suggestion
            1. Parse _P_ using the grammars in <emu-xref href="#sec-patterns"></emu-xref>. The goal symbol for the parse is |Pattern[+U, +N]|. If _P_ did not conform to the grammar, if any elements of _P_ were not matched by the parse, or if any Early Error conditions exist, return *false*.
```